### PR TITLE
 Locust integration sumaform

### PR DESCRIPTION
**Pull request by MalloZup**
TODOs:

- [x] leap42.3 support
- [x] Add simple example inside
- [x] Add documentation

- [x] test openstack backport

- [x] add runner for prometheus which run the exporter and start locust

### CODE REVIEW

**Comment by MalloZup** [comment]
@moio before i make the doc, you can have a look.

Basically this role setup the host for running locust. We inject the yaml file to spacewalk and then the pipeline we schedule the run with the appropriate locust command. 
```hcl
module "srv" {
  source = "./modules/libvirt/suse_manager"
  version = "head"
  base_configuration = "${module.base.configuration}"
  name = "srvmgr"
  image = "sles12sp3"
  for_development_only = true
  for_testsuite_only = false
}

module "locust" {
  source = "./modules/libvirt/locust"
  base_configuration = "${module.base.configuration}"
  server_configuration = "${module.srv.configuration}"
  git_username = "foo"
  git_password = "bar"
}

```

**Comment by MalloZup** [comment]
@moio  ok thx.

I fixed the functionality for injecting a custom locust_file: https://github.com/moio/sumaform/pull/260/files#diff-029d9a2231362d5c78164a87e4d7646bR76

File: modules/libvirt/locust/main.tf [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+mirror: ${var.base_configuration["mirror"]}
+server: ${var.server_configuration["hostname"]}
+role: locust
+git_username: ${var.git_username}
+git_password: ${var.git_password}
+locust_file: ${base64encode(file(var.locust_file))}
+EOF
+
+  // Provider-specific variables
+  image = "opensuse423"
```
Comment (moio) [inline modules/libvirt/locust/main.tf]: I forgot about this: we need opensuse repos in the mirror. Please open at least an issue (even better: a follow up PR to edit the minima file)

File: salt/locust/init.sls [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  pkg.installed:
+    - pkgs:
+      - gcc
+      - python-devel
+      - python-pyzmq-devel
+      - git-core
+
+spacewalk_git_repository_locust:
+  cmd.run:
+    - name: git clone --depth 1 https://github.com/SUSE/spacewalk -b Manager /root/spacewalk
```
Comment (moio) [inline salt/locust/init.sls]: To me locust tests should not be pulled from GitHub at all, they should just be specified via `locust_file`. Whether that's pulled from GitHub or anywhere else is a separate problem to me.

File: salt/locust/init.sls [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    - user: root
+    - group: root
+    - mode: 755 
+    - force: True
+    - require:
+      - cmd: spacewalk_git_repository_locust
+
+locust_config_file_for_minimal_example:
+  file.managed:
+    - name: /root/locust_config.yml
```
Comment (moio) [inline salt/locust/init.sls]: I would avoid replicating this in two places - what is the reason not to use one only?

Comment (MalloZup) [reply salt/locust/init.sls]: if we are not going to clone spacewalk from the locust node, we need a locust_config.yml file which is needed for the minimal example

File: salt/locust/init.sls (hunk 2 above)
Comment (MalloZup) [reply salt/locust/init.sls]: ok

File: modules/libvirt/locust/main.tf (hunk 1 above)
Comment (MalloZup) [reply modules/libvirt/locust/main.tf]: i will create an issue, for bonus points i will fix it, but i am on other stuff atm

Comment (MalloZup) [reply modules/libvirt/locust/main.tf]: https://github.com/moio/sumaform/issues/264

File: README_ADVANCED.md [hunk 4]
```diff
@@ -439,11 +439,27 @@ module "evil-minions" {
   base_configuration = "${module.base.configuration}"
 
   name = "evil-minions"
-  server_configuration = "${module.suma31pg.configuration}"
+  server_configuration = "${module.srv.configuration}"
```
Comment (MalloZup) [inline README_ADVANCED.md]: removing version of servers with generic one (nitpick)

File: salt/locust/locust-exporter.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/python
```
Comment (MalloZup) [inline salt/locust/locust-exporter.py]: on all file

```python

#!/usr/bin/env python
```

File: salt/locust/locustfile.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
+#! /usr/bin/python
```
Comment (MalloZup) [inline salt/locust/locustfile.py]: same as above

File: salt/locust/run-locust.py [hunk 7]
```diff
@@ -0,0 +1,21 @@
+#! /usr/bin/python
```
Comment (MalloZup) [inline salt/locust/run-locust.py]: nitpick: #!/usr/bin/env python

**Comment by MalloZup** [comment]
@moio  to me seems correct the patch, just a small nitpick from my pov

**Comment by moio** [comment]
Hey @MalloZup, thanks for another awesome addition to the codebase!

**Pull request merged**